lib/pry/history.rb

```diff
@@ -78,8 +80,26 @@ def to_a
       @history.dup
     end
 
+    # Filter the history with the histignore options
+    # @return [Array<String>] An array containing all the lines that are not included
+    # in the histignore.
+    def filter(history)
+      history.select { |l| l unless should_ignore?(l) }
```

Comment:
:+1: @bronzdoc `select` is appropriate here, alternatively you could use `reject` and invert the predicate, but no big deal here